vowpalwabbit/vw_validate.cc

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright (c) by respective owners including Yahoo!, Microsoft, and
+individual contributors. All rights reserved.  Released under a BSD (revised)
+license as described in the file LICENSE.
+*/
+
+#pragma once
+
+#include "global_data.h"
+#include "vw_validate.h"
+#include "vw_versions.h"
+
+namespace VW
+{
+    void validate_version(vw& all)
+    {
+        if (all.model_file_ver < LAST_COMPATIBLE_VERSION || all.model_file_ver > PACKAGE_VERSION)
+        {
+            THROW("Model has possibly incompatible version! " << all.model_file_ver.to_string());
+        }
+    }
+
+    void validate_unexpected_eof(size_t nbytes)
```

Comment:
I can rename this. The check is not supposed to specialize to model only, although the error message certainly needs updating.